Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -231,8 +265,8 @@ open class RouteController: NSObject {
     @objc private func navigationStatusDidChange(_ notification: NSNotification) {
         guard let userInfo = notification.userInfo,
               let status = userInfo[Navigator.NotificationUserInfoKey.statusKey] as? NavigationStatus else { return }
-        DispatchQueue.main.async { [weak self] in
-            self?.update(to: status)
+        onMainQueueSync {
```

Comment:
`navigationStatusDidChange` is called on the thread that create NavNative navigator which is most likely to be the main thread, but not guaranteed at the moment. Solutions:
- `onMainQueueSync`.
- Put preconditions that navigator must be created on the main thread.
- Enforce the creation of NavNative navigator on the main thread. 

I think the best solution would be " Enforce the creation of NavNative navigator on the main thread".